src/Tools/dotnet-monitor/CollectionRules/CollectionRulePipeline.cs

```diff
@@ -226,5 +229,15 @@ public new Task StopAsync(CancellationToken token)
         {
             return base.StopAsync(token);
         }
+
+        // Ensures that the start callback is only invoked once.
+        private void InvokeStartCallback()
+        {
+            if (_invokedStartCallback)
```

Comment:
Should this be negated? As in, if we haven't invoked the callback, we do it (and set it to true) so we can only invoke it once?

Reply:
Good catch; I'll adjust the tests to make sure this does in fact work after fixing it.